Userland/Libraries/LibWeb/HTML/Canvas/CanvasPath.cpp

```diff
@@ -213,4 +213,154 @@ void CanvasPath::rect(double x, double y, double w, double h)
     m_path.move_to(transform.map(Gfx::FloatPoint { x, y }));
 }
 
+// https://html.spec.whatwg.org/multipage/canvas.html#dom-context-2d-roundrect
+WebIDL::ExceptionOr<void> CanvasPath::round_rect(double x, double y, double w, double h, Optional<Variant<double, Geometry::DOMPointInit, Vector<Variant<double, Geometry::DOMPointInit>>>> radii)
+{
+    // 1. If any of x, y, w, or h are infinite or NaN, then return.
+    if (!isfinite(x) || !isfinite(y) || !isfinite(w) || !isfinite(h))
+        return {};
+
+    // 2. If radii is an unrestricted double or DOMPointInit, then set radii to « radii ».
+    if (radii->has<double>() || radii->has<Geometry::DOMPointInit>()) {
+        Vector<Variant<double, Geometry::DOMPointInit>> radii_list;
+        if (radii->has<double>())
+            radii_list.append(radii->get<double>());
+        if (radii->has<Geometry::DOMPointInit>())
+            radii_list.append(radii->get<Geometry::DOMPointInit>());
+        radii = radii_list;
+    }
+
+    // 3. If radii is not a list of size one, two, three, or four, then throw a RangeError.
+    if (radii->get<Vector<Variant<double, Geometry::DOMPointInit>>>().is_empty() || radii->get<Vector<Variant<double, Geometry::DOMPointInit>>>().size() > 4)
```

Comment:
I'd be tempted to add a `using InsertNameHere = Vector<Variant<double, Geometry::DOMPointInit>>` somewhere, because this gets written a few times, but it's up to you.